internal/bminventory/inventory.go

```diff
@@ -2181,6 +2167,54 @@ func (b *bareMetalInventory) UploadHostLogs(ctx context.Context, params installe
 	return installer.NewUploadHostLogsNoContent()
 }
 
+func (b *bareMetalInventory) DownloadHostLogs(ctx context.Context, params installer.DownloadHostLogsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Downloading logs from host %s in cluster %s", params.HostID, params.ClusterID)
+	hostObject, err := b.checkHostLogs(params.ClusterID.String(), params.HostID.String())
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+
+	fileName := b.getLogsFullName(params.ClusterID.String(), params.HostID.String())
+	exists, err := b.s3Client.DoesObjectExist(ctx, fileName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get host %s logs", params.HostID)
+		return common.GenerateErrorResponder(common.NewApiError(http.StatusInternalServerError, err))
+	} else if !exists {
+		log.Warnf("Failed to find %s in s3", fileName)
+		return common.NewApiError(http.StatusNotFound, errors.Errorf("Logs for host %s not found", params.HostID))
+	}
+
+	respBody, contentLength, err := b.s3Client.Download(ctx, fileName)
+	if err != nil {
+		log.WithError(err).Errorf("failed to download file %s", fileName)
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	downloadFileName := fmt.Sprintf("%s_%s", common.GetHostnameForMsg(hostObject), filepath.Base(fileName))
+	return filemiddleware.NewResponder(installer.NewDownloadHostLogsOK().WithPayload(respBody), downloadFileName, contentLength)
+}
+
+func (b *bareMetalInventory) getLogsFullName(clusterId string, hostId string) string {
+	return fmt.Sprintf("%s/logs/%s/logs.tar.gz", clusterId, hostId)
+}
+
+func (b *bareMetalInventory) checkHostLogs(clusterId string, hostId string) (*models.Host, error) {
```

Comment:
1. The function name isn't a good one. This function only does `getFirstHostOfCluster`. So I'm not sure if this is what you tried to achieve but at least change the function name. Because it doesn't relate to logs.
2. In case you want to receive the same host - you don't take care if the 1st host of the Cluster isn't the one you are looking for. You should go in a loop

Reply:
This function validates that cluster with given id exists and that given host id exists in this cluster. For not to have same code twice it return host object for further usage. It will be the only host and not the first one.

Reply:
So call it assertHostExists and we can use it elsewhere

Reply:
possible

Reply:
ok